src/main/scala/cromwell/binding/values/WdlFile.scala

```diff
@@ -9,9 +9,19 @@ object WdlFile {
     if (path1.endsWith("/") || path2.startsWith("/")) path1 + path2
     else path1 + "/" + path2
   }
+
+  def apply(value: String, isGlob: Boolean = false): WdlFile = {
+    if(isGlob) {
+      WdlGlobFile(value)
+    } else {
+      WdlSingleFile(value)
+    }
+  }
 }
 
-case class WdlFile(value: String) extends WdlPrimitive {
+sealed trait WdlFile extends WdlPrimitive {
+  val value: String
+  val isGlob: Boolean
```

Comment:
Thinking about this, there's probably not even a reason for this to even be a boolean any more. All the information is already in the type.